src/main/java/com/solace/samples/java/patterns/DirectReplierNonBlocking.java

```diff
@@ -0,0 +1,122 @@
+        if (args.length < 3) {  // Check command line arguments
+            System.out.printf("Usage: %s <host:port> <message-vpn> <client-username> [password]%n%n", SAMPLE_NAME);
+            System.exit(-1);
+        }
+        System.out.println(API + " " + SAMPLE_NAME + " initializing...");
+
+        //2. Set up the properties including username, password, vpnHostUrl and other control parameters.
+        final Properties properties = new Properties();
+        setupPropertiesForConnection(properties, args);
+
+        //3. Create the MessagingService object and establishes the connection with the Solace event broker
+        final MessagingService messagingService = MessagingService.builder(ConfigurationProfile.V1).fromProperties(properties).build();
+        messagingService.connect();  // blocking connect to the broker
+
+        //4. Register event handlers and callbacks for connection error handling.
+        setupConnectivityHandlingInMessagingService(messagingService);
+
+        //5. Build and start the Receiver object
+        RequestReplyMessageReceiver requestReplyMessageReceiver = messagingService.requestReply().createRequestReplyMessageReceiverBuilder().build(TopicSubscription.of(TOPIC_PREFIX + "*/direct/request"));
+        requestReplyMessageReceiver.start();
+        //5-A. Set up an event handler for situations where the reply message could not be published.
+        requestReplyMessageReceiver.setReplyFailureListener(failedReceiveEvent -> System.out.println("### FAILED RECEIVE EVENT " + failedReceiveEvent));
+
+        //6. Create an OutboundMessageBuilder for building the outbound reply message
+        final OutboundMessageBuilder outboundMessageBuilder = messagingService.messageBuilder();
+
+        //7. Define the handler for the incoming message.
+        final RequestReplyMessageReceiver.RequestMessageHandler messageHandler = (inboundMessage, replier) -> {
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+//          System.out.println("The inbound message is : " + inboundMessage.dump()); // Enable this for learning purposes as it logs a String representation of the whole Message
+            final String stringPayload = inboundMessage.getPayloadAsString();
+            System.out.println("The converted message payload is : " + stringPayload);
+
+            //7-B. Create the outbound message payload.
+            final StringBuilder outboundMessageStringPayload = new StringBuilder().append("Hello! Here is a response to your message on topic : ").append(inboundMessage.getDestinationName()).append(" with incoming payload :").append(stringPayload);
+
+            //7-C. Create the outbound message with headers and payload.
+            final OutboundMessage outboundMessage = outboundMessageBuilder.build(outboundMessageStringPayload.toString());
```

Comment:
is `toString()` necessary if `outboundMessageStringPayload` is already a new StringBuilder? Not sure if this is a Java specific approach

Reply:
So calling a toString on StringBuilder is the way to extract the String content from it. 
Simply passing the StringBuilder object wont work with the API and its a type mismatch.
![image](https://user-images.githubusercontent.com/112864703/201646990-07aee409-2d16-47f4-81dc-61d05992f2c5.png)